package/kernel/mac80211/patches/ath9k/552-ath9k-ahb_of.patch

```diff
@@ -162,27 +162,25 @@
 +
 +static int of_ath_ahb_probe(struct platform_device *pdev)
 +{
-+	struct ath9k_platform_data *pdata;
++	struct ath_hw *ah = platform_get_drvdata(pdev);
 +	const struct of_device_id *match;
 +	const struct of_ath_ahb_data *data;
 +	u8 led_pin;
 +
 +	match = of_match_device(of_ath_ahb_match, &pdev->dev);
 +	data = (const struct of_ath_ahb_data *)match->data;
 +
-+	pdata = dev_get_platdata(&pdev->dev);
-+
 +	if (!of_property_read_u8(pdev->dev.of_node, "qca,led-pin", &led_pin))
-+		pdata->led_pin = led_pin;
++		ah->led_pin = led_pin;
```

Comment:
This comment unrelated to this PR. We have some discussions several months ago about this `qca,led-pin` property. After that, I tested it on QCA9531 and AR9287, but it didn't work at all.

Reply:
Correct. It works on mpc85xx but on ath79 there’s an EPROBE_DEFER that gets thrown and not handled.

Reply:
From gpio_request_one IIRC